File: packages/ts/src/a2a/text.ts
/**
 * Shared sanitiser for Agent Card extension strings. Card text travels from an agent we do not
 * control straight to the registry and, for the showcase profile, to a rendered page — so control
 * characters and angle brackets are stripped and whitespace is collapsed before anything else.
 */
export function compactText(value: string): string {
  return Array.from(value, (character) => {
    const code = character.charCodeAt(0)
    return code < 32 || code === 127 ? ' ' : character
  })
    .join('')
    .replace(/[<>]/g, ' ')
    .replace(/\s+/g, ' ')
    .trim()
}

File: packages/python/src/ai37_agent_sdk/a2a/text.py
from __future__ import annotations


def compact_text(value: str) -> str:
    """Санитайзер строк карточки агента: карточку пишет агент, которым мы не управляем, а витрина
    печатает её текст пользователю — поэтому убираем управляющие символы и угловые скобки, а
    пробелы сводим к одному."""
    safe = "".join(
        " " if ord(character) < 32 or ord(character) == 127 else character for character in value
    )
    return " ".join(safe.replace("<", " ").replace(">", " ").split())
